pkg/cliplugins/workspace/plugin/kubeconfig.go

```diff
@@ -406,3 +406,52 @@ func (kc *KubeConfig) ListWorkspaces(ctx context.Context, opts *Options) error {
 
 	return printer.PrintObj(table, opts.Out)
 }
+
+func (kc *KubeConfig) CreateContext(ctx context.Context, name string) error {
+	config, err := clientcmd.NewDefaultClientConfig(*kc.startingConfig, nil).RawConfig()
+	if err != nil {
+		return err
+	}
+	currentContext, ok := config.Contexts[config.CurrentContext]
+	if !ok {
+		return fmt.Errorf("current context %q is not found in kubeconfig", config.CurrentContext)
+	}
+	currentCluster, ok := config.Clusters[currentContext.Cluster]
+	if !ok {
+		return fmt.Errorf("current cluster %q is not found in kubeconfig", currentContext.Cluster)
+	}
+	_, currentClusterName, err := parseClusterURL(currentCluster.Server)
+	if err != nil {
+		return fmt.Errorf("current URL %q does not point to cluster workspace", currentCluster.Server)
+	}
+
+	if name == "" {
+		name = currentClusterName.String()
+	}
+
+	_, existedBefore := kc.startingConfig.Contexts[name]
+
+	newKubeConfig := kc.startingConfig.DeepCopy()
+	newCluster := *currentCluster
+	newKubeConfig.Clusters[name] = &newCluster
+	newContext := *currentContext
+	newContext.Cluster = name
+	newKubeConfig.Contexts[name] = &newContext
+	newKubeConfig.CurrentContext = name
+
+	if err := kc.modifyConfig(newKubeConfig); err != nil {
+		return err
+	}
+
+	if existedBefore {
+		if kc.startingConfig.CurrentContext == name {
+			fmt.Fprintf(kc.Out, "Updated context %q.\n", name) // nolint: errcheck
+		} else {
+			fmt.Fprintf(kc.Out, "Updated context %q and switched to it.\n", name) // nolint:	errcheck
```

Comment:
Should we ask the user if they want to overwrite a preexisting non-current context, or make them specify `--overwrite`? e.g. I have a context called `prod`, I switch my current workspace to `test`, and then do `k kcp create-context prod`, I probably don't actually want to do that?

Reply:
Flag seems to be fine. Has there been anything interactive in kubectl and oc before?

Reply:
I would do what `kubectl label` does:

```
 If --overwrite is true, then existing labels can be overwritten, otherwise attempting to overwrite a label will
result in an error.
```